### ISSUE TITLE
"Error Decrypting" messages (note plural) -- OK now what?

### ISSUE THREAD

**Issue by WritingSolutions2020** [standardnotes/forum#41]
Now I am [annoyed ] for real. I've tried to be patient but this is COMPLETELY UNACCEPTABLE!

I just used the Web client. I didn't intentionally do anything wacky or weird. And now  I have a whole bunch of "Error Decrypting" messages:

Web client:

![2017-07-27 12_04_02-notes standard notes](https://user-images.githubusercontent.com/3333327/28687795-7ce2f5a8-72c4-11e7-8fc0-80ac910f038b.png)

W7 desktop client:

![2017-07-27 12_05_13-notes](https://user-images.githubusercontent.com/3333327/28687836-a9f021e2-72c4-11e7-9019-c1a05456eb0d.png)

What the Hell does this even mean!? And what am I supposed to do about it!?

I've been using computers since 1981. I was a former systems programmer who wrote assembly language and C++. I specialized in hardware-software integration. Today I write my own Python 3 utilities -- I understand how computers work! And as a "technical technical writer since 1998 I can work around minor annoyances. This is NOT a minor annoyance!

This is crazy! Sorry if I seem annoyed. But I am. What program should "offer" me this information? Moreover, what program should offer me this information and provide me with ZERO guidance as to what to do next!?

C'mon...

Hope this helps,
Riley
SFO

**Comment by mobitar** [comment]
See https://github.com/standardnotes/forum/issues/24

**Comment by WritingSolutions2020** [comment]
I'm sorry my friend but this is a train wreck! How many steps -- EXACTLY -- do I need to perform?

No, no, no, no, no! There is some evidence to suggest that I am neither stupid nor lazy. I've been using computers since 1981. I am a former software developer who (way back in 1985) used assembly language to write interrupt handlers and to integrate new hardware with operating systems. And I am now genuinely [annoyed].

Presenting users with scary and unhelpful messages like the ones illustrated by my screenshots is bad UX (as in, what should I do now?). Suggesting that users confronted with such messages are now obliged to do some mysterious set of steps to remediate the problem is worse.

Please understand that if Standard Notes is to succeed, it cannot expect users to fix application problems by reading github bug reports...

Cheers, thanks as always, and hope this helps,
Riley
SFO

**Comment by WritingSolutions2020** [comment]
What I _seem_ to be able to do  is delete all the error instances(?) I did so tentatively -- I worried that deleting those objects would also delete the notes I don't want to lose.

Again, I'm not eager to be a complainer or to be negative. But as described -- and experienced -- this is a [very poor] UX..

Cheers, thanks as always, & hope this helps,
Riley
SFO

**Comment by mobitar** [comment]
I'm not following..did the steps fix the issue?

**Comment by WritingSolutions2020** [comment]
> "did the steps fix the issue?"

Logging out and back in didn't seem to. In the Web client I logged out, deleted my browser's cache and all Standard Notes cookies. then logged back in. The "Error decrypting" notes were all still there. (I say  "notes"? because as the steps I used to get rid of those self-described errors were exactly the same steps I I use to delete a note.

Which is to say that what I did git rid of all these self-described errors was: Selected each one of the "Error decrypting" instances one-by-one, then in each instance / note click **Menu** > *Delete note**. Now if I log out and back in I seem to have cleared the problem in both the Web and the desktop client.

From my perspective the MUCH more important point is: _What_ innocent / unintentional hand movement did this normally skilled user make to create the situation? I wasn't "fooling around" or "experimenting" or otherwise doing some "not-normal-user" thing. Then shazam! -- I suddenly have a whole bunch of "things" (notes? or?) that report "Error decrypting". Ouch! Sigh...

Cheers, thanks as always, & hope this helps,
Riley
SFO

**Comment by WritingSolutions2020** [comment]
OK -- what I have evidently ended up losing was ALL my tags. They're all gone. Where? I don't know...

I guess it could have been worse: I might have lost all my notes.

Sigh...

**Comment by mobitar** [comment]
By deleting each note individually, even when there was an error decrypting it, you delete it completely from the server. So at this point you've deleted notes that were otherwise valid.

The reason this happened is likely because you applied the Verification tag update, but did not update your applications or sign out afterwards.

Regarding your tags, I think they still exist, but that they don't show up using the Folders extension because there was an error decrypting them. Try deactivating the Folders extension. At this point, you'll see a list of tags with a blank title (because they aren't decrypted).

I understand your frustration. We'll get to the bottom of this. No need to worry though, your data is never lost. I can restore your data to one of the backups made every morning. Shoot me an email with your account email to mo@standardnotes.org.

Answer a few questions as well if you can:
1. Have you changed your password recently?
2. Did you upload a verification tag to your account recently?
3. Have you created any backups of your data manually?

**Comment by mobitar** [comment]
Also, while the Verification update has been troublesome for a handful of users, it only required manual migration for existing users, and not new users. Luckily new users get all of that automatically. But it was a migration that had to be done. I knew it would cause temporary pain and bad UX for some users, but it was one of those bandages that just needed to be ripped off all at once.

**Comment by AdamFo** [comment]
He,y @mobitar.
I am the new user. I've downloaded windows desktop app and iPhone app TODAY, and registered TODAY for the FIRST time.

I have a similar problem. When I make a note in iPhone it shows me "error decrypting" message on my desktop app version. AND when I make a desktop note it shows me "error decrypting" message on my IPhone.

I hope this issue will get done fast, because I am eager to try this app!

**Comment by mobitar** [comment]
@AdamFo continued here: https://github.com/standardnotes/forum/issues/42

**Comment by WritingSolutions2020** [comment]
Mo:

Per our question, no -- didn't change my password. The only change I made was that I added the extension that attaches a tag to a Note based on typed characters.

More the point, it looks like I've been able to tidy things up. Whew! Scary.

Essential point is that a whole bunch of invalid objects somehow came into being.

* In the list of Notes I had the aforementioned *Error decrypting* objects. These behaved like Notes so I deleted them one-by-one. These are only "Notes" I seem to have "lost", albeit intentionally.

* In the *Extensions* area there were  (yet again) multiple instances of the *Action bar* extension. I deleted all but one. (Don't know why the *Action bar* wants to create multiple instances of itself. This was a problem I had per an earlier issue report; I thought I'd put that right. But here again...) I then disabled the last instance; at least for the time being.

* In the *Extensions* area there were several "blank bars" that behaved like editor extensions when clicked. I clicked then deleted any of them that didn't have text on them.

* In the *Extensions* area I removed the extension that attaches tags to notes. No way to know if that was the source of my problem. But I'm staying away from it for now.

* I again logged out of the desktop client, then back in.

OK -- things look like they're back the way the way they were. No apparent loss of Notes or Tags (he said).

Scary. I wasn't doing anything weird save adding that tags extension. And I can't say adding it was the source of the problem -- may well have been a coincidence. With that in mind, I have no idea why the application suddenly became that frighteningly confused.

Cheers, thanks, & hope this helps,
Riley

**Comment by mobitar** [comment]
So per #42, I just uncovered an issue where if you create notes offline on iOS (and possibly Android), and then sign in to your account, those notes then become problematic when you sync them to desktop. Did you happen to use any mobile clients during this time?

Or did you happen to create any notes **offline** in general during this time?

**Comment by WritingSolutions2020** [comment]
Re:

> Did you happen to use any mobile clients during this time?

> ...did you happen to create any notes offline in...

* None of the above. I don't have a mobile client: Firefox 53 browser or Windows client, only.
* I didn't (intentionally) create any offline notes; not even sure how I'd do that.

That's why the whole experience was so scary. I wasn't doing anything I would suppose would "stress" the application. I just started the desktop for Windows client and all Hell broke loose...

Cheers and thanks,
Riley
SFO

**Comment by mobitar** [comment]
And were you already signed in on both clients when the issue happened? Or did you initiate a new sign in?

**Comment by WritingSolutions2020** [comment]
Re: 

> ...were you already signed in on both clients when the issue happened?

Maybe yes. Not sure. I usually only turn to the Web client if I need to add an extension. So because I was adding an extension I had to have the Web client live.

Not sure if I then fired up the desktop client, too. 

Cheers & hope this helps,
Riley

**Comment by mobitar** [comment]
Here's what I think happened. Let me know if this sounds plausible.

You've been using the latest version of the desktop client, and everything has been going great. You applied the Verification update sometime last week or so.

You come to the web app after a while, and you're already signed in. However, because you're already signed in and haven't visited in a while, the web browser doesn't have the Verification parameters, and so is unable to decrypt all your items. (Strangely, now that I take a look at your screenshot, all the items with error decrypting have the same date and time. It may have just been one item that was duplicated as part of some sync conflict).

At this point, the only solution would have been to sign out of the web app and sign back in. However, because it wasn't clear what you should do, you begin deleting things, which takes us into a whole other stranger land of confusion.

Does that sound possible?

**Comment by WritingSolutions2020** [comment]
Mo:

> You come to the web app after a while, and you're already signed in.
> However, because you're already signed in and haven't visited in a while, 
> the web browser doesn't have the Verification parameters,

Possibly. For what it's worth, I use a cleanup application that deletes browser cookies and the browser cache around at the end of a browser session. And the start of a browser session corresponds to my booting up my laptop at the start of each day; as in I don't leave my laptop running overnight. Point being that if I log in to anything at the start of a browser session a Web app has no residual cookies or cache.

More than anything I was overwhelmed by a sudden explosion of red ink *Error decrypting* messages. Accompanied by apparent chaos in the Extensions panel. And my initial attempts to click my way into an understanding of what the heck happened, and to then click my way into a fix, didn't initially produce a helpful outcome. Scary! And yes, annoying.

In any event I seem to have gotten things back where they were. Next time (I suspect there will, alas, be one) I'll have a clearer idea of what to try...

Cheers, thanks for your help & patience,
Riley
SFO

**Comment by mobitar** [comment]
I'm more optimistic that you won't experience something like this again ;) Given the timing, it's 100% likely this is related to the Verification update as I've had to deal with dozens of similar issues in the past few days.

In any case, I'll be releasing a new service in the next couple weeks that auto-emails encrypted backups of your data to you every night. That should help for people who don't have Dropbox or Google Drive.

**Comment by WritingSolutions2020** [comment]
Web client via MintLinux 32-bit with Firefox 54, **logged in to Standard Notes for the first time on this computer.**

Here we go again:

![screenshot at 2017-07-30 15-11-20](https://user-images.githubusercontent.com/3333327/28757554-76ee0e9e-753a-11e7-8380-2b95cab903fb.png)

Not sure why this recurred given that this is the FIRST time I logged in to Standard Notes from this computer with this browser. So the notion of a "Verification update" doesn't seem like it should apply?

What a poor user experience. I did nothing "wrong" (except perhaps try to use Standard Notes) but I'm again obliged to sort this out. Note also that at this point I'm tired of documenting this category of problem...

Please fix this forever! Please don't do this to your users! For future updates or changes that risk creating this category of mess, please find a way to make it less frightening -- and easier to clean up...

**Comment by mobitar** [comment]
Can you recall what you did leading up to this? Because it seems that your actual data is decrypted properly, but that the app is creating a bunch of duplicates of some invalid item all with the same timestamp: 7/30/2017 3:00PM. What about notes prior to that? Are they decrypting properly?

And can you speak of the note of 7/30/2017 3:00PM? Did you create a note at this time?

**Comment by mobitar** [comment]
Also, what happens if you open an incognito tab on a web browser and try a fresh sign in from there? Do those same items show that same error?

**Comment by WritingSolutions2020** [comment]
> Can you recall what you did leading up to this?

In MintLinux 32-bit I used Firefox 54 to log into Standard Notes. Was immediately greeted with the red ink. Repeated the manual select and delete.

Thanks for your help and patience but this is my last comment on this issue. If it happens again on any system or client, I'm gone...

Good luck!

**Comment by mobitar** [comment]
What about notes prior to 7/30/2017 3:00PM? Are they decrypting properly?

**Comment by WritingSolutions2020** [comment]
> Are they decrypting properly?

How would I know? I evidently haven't lost my notes. Does that mean "decrypting properly"?

I'm not doing anything more to troubleshoot this --I'm exhausted. If Standard Notes "blows up" again I'm simply going to walk away from it...

**Comment by mobitar** [comment]
Discovered what the issue is. Will report back once I have the fix.

**Comment by mobitar** [comment]
Ok, go ahead and give Desktop v1.2.2 a download (or auto-update if not on Linux — auto-update n/a on Linux) and refresh the web app to get the latest version. Mobile clients will be updated with the fix this week. This should solve your issue.

If it doesn't, then 😃🔫.

**Comment by WritingSolutions2020** [comment]
OK, thanks -- installed. Here's hoping -- there are several things I like about Standard Notes *when it works*. But major explosions of the sort described in this thread are truly off-putting...

Cheers & thanks 'gain for your help,
Riley
SFO

**Comment by perandre** [comment]
Seeimg the same issue: Opening the iOS app (latest version) for the first time in weeks, all notes are replaced with the Error Decrypting msg. Tried to sign out and sign in again, but it didn’t help.

**Comment by mobitar** [comment]
Did you restore from iCloud recently? There was an issue with this that has since been fixed. Please email me from your account email to mo@standardnotes.org and I'll help you get this resolved.

**Comment by perandre** [comment]
No, no restore has been done. Thanks, I’ll email you now.

**Comment by Stever888** [comment]
Having this problem, too, after an iCloud restore to initialize a replacement iPhone.

**Comment by Stever888** [comment]
(HELP!!!) in case that wasn’t obvious

**Comment by mobitar** [comment]
Resolution summary:

[perandre's issue](https://github.com/standardnotes/forum/issues/41#issuecomment-354020619): In this case the user's account used an old encryption format that the new app was for some reason unable to read. The solution was to use the web/desktop and follow the update prompts upon signing in. Then click Account -> Advanced -> Re-encrypt all items.

[Stever888's issue](https://github.com/standardnotes/forum/issues/41#issuecomment-354905753): User was able to see notes properly decrypt after verifying that the data was intact using the web app first, then (to the effect of) deleting the iOS app and re-installing it to have a safer, blank slate.

However, while I was digging around in the codebase, I identified a deeper issue with the iCloud restore bug, which I thought I had fixed. The issue can still manifest itself if the user accidentally does the following:

1. Restore from iCloud.
2. Open the app, see "Error decrypting".
3. Quit the app completely.
4. Re-open the app and sign in.

In this case, the "Error decrypting" flag was not saved properly after the user quits the app, so upon re-open, the items forget that they are in an errored state, and upon signing in, are merged with the user's data.

Fortunately I was able to [patch](https://github.com/standardnotes/mobile/commit/47d81d109ce393a5feab2e570072f9fec4d4fe1e) this up quickly, and will be rolling out the fix this week for all platforms.
